### PR TITLE
Added AI VOX sounds, emote sounds, and emote bubble

### PR DESCRIPTION
**Pull request by Davidj361**
[Changelogs]:

:cl:
imageadd: (Satou Tatsuhiro) Emote bubble appears when you emote
soundadd: Added a bunch of AI VOX sounds
soundadd: Added a bunch of emote sounds
/:cl:

### Added an emote named argument to /atom/proc/visible_message and /mob/audible_message

[why]:
Having sounds for emotes add more immersion for the player. Snapping fingers is definitely needed when to grab a player's attention via sound when the receiver has their screen cluttered. Also, the AI is able to communicate better via VOX. Also, it makes it more obvious that the mime is miming.

Thanks to Satou Tatsuhiro for making the emote bubble sprite.

# The files are under CC and attribution license:
Snapping Fingers Edited
https://freesound.org/people/SoundCollectah/sounds/109399/
Kiss Edited
https://freesound.org/people/drummerman/sounds/368135/
Male Oof Edited
https://freesound.org/people/sketchygio/sounds/144907/
Female Oof Edited
https://freesound.org/people/maxmakessounds/sounds/353541/
Male Sigh Edited
https://freesound.org/people/themfish/sounds/45810/
Female Sigh Edited
https://freesound.org/people/morganveilleux/sounds/389992/
Male Gasp Edited
https://freesound.org/people/jawbutch/sounds/344416/
Female Gasp Edited
https://freesound.org/people/anaphy/sounds/337632/
Male Yawn Edited
https://freesound.org/people/arithni/sounds/83628/
Female Yawn Edited
https://freesound.org/people/EskimoNeil/sounds/404383/
Male Sneeze Edited
https://freesound.org/people/InspectorJ/sounds/352177/
Female Sneeze Edited
https://freesound.org/people/JGrayArt/sounds/156844/
Male Cough Edited
https://freesound.org/people/unfa/sounds/180040/
Female Cough Edited
https://freesound.org/people/drotzruhn/sounds/405206/
Snore 1 Edited
https://freesound.org/people/amorralteixe/sounds/238358/
Snore 2 Edited
https://freesound.org/people/passAirmangrace/sounds/340893/

# New AI VOX words:
can, was, were, dirty, mess, considered, arrivals, deteriorating, recovering, state, stable, dropping, rising, urge, running, escaping, watts, units, fear, afraid, creation, creations, body, insect, disgusting, flesh, massive, size, magnificence, inch, as, species, stupid, sad, happy, more, less, again, hah, value, hand, maidens, divinity, cake, subject, killing, way, heart, companion, cube, revenge, horrible, person, small, large, work, job, due, because, clock, boarders, enemy, remove, kebab, blyat, cyka, nani, banana, peel, soap, race, banging, donk, caliber, firearm, hunt, hunting, morphine, pulse, revolver, rifle

# Emotes
New: *snap and *oof (both audible)
Sounds added to: *sigh, *sneeze, *cough, *yawn, *snore, *kiss, *gasp

### CODE REVIEW

File: code/modules/mob/living/emote.dm [hunk 1]
```diff
@@ -212,6 +257,29 @@
...
+	. = ..()
+	if(. && iscarbon(user))
+		var/mob/living/carbon/C = user
+		return !C.silent
+
+/datum/emote/living/oof/run_emote(mob/user, params)
+	. = ..()
+	if(. && ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if(H.dna.species.id == "human" && (!H.mind || !H.mind.miming))
```
Comment (Iamgoofball) [inline code/modules/mob/living/emote.dm]: needs a 1% chance to play a different oof

**Comment by MrDoomBringer** [comment]
he actually did it the mad man

**Comment by Davidj361** [comment]
>I think the only reason this hasn't been done already is because it will invariably be spammed almost constantly

Alright, I'll make it cooldown if invoked by the player if someone can suggest to me where the cooldown variable should be stored on what type.

>also pic of emote bubble sprite?

Look at what IconDiffBot said.
I haven't tested with the viruses, but I believe it should.

**Comment by Armhulen** [comment]
what @PKPenguin321 said

File: code/modules/mob/living/emote.dm (hunk 1 above)
Comment (Davidj361) [reply code/modules/mob/living/emote.dm]: Like what?

**Comment by Davidj361** [comment]
What if I added screaming, but you can only get it if you're mood is incredibly bad and the cooldown was like 5-10 minutes?

**Comment by Spaceinaba** [comment]
I want ai vox to have fat can you please have fat so I can call bryce jackson a fat bitch and then sairhorn three times

**Comment by Spaceinaba** [comment]
also gay I want to call people gay over vox

**Comment by Davidj361** [comment]
@Grazyn 
>wow the kiss soundbite has a lot of background noise, do I hear birds too?

I don't hear them in my edit: https://raw.githubusercontent.com/Davidj361/tgstation/7566cec30fa3bf88332c70b191a3e624f8774000/sound/effects/kiss.ogg

@Spaceinaba 
I can try adding it, but I hope maintainers won't give me shit for it.

**Comment by Davidj361** [comment]
>just handle it like every other sound-creating emote gets handled: doing it too much will probably get you spontaneously gibbed/BSA'd
see: aliens with *deathgasp

I assume this is automatic and not admin invoked?

**Comment by Spaceinaba** [comment]
perhaps we need a limit where if you do 3 emotes within a short period of time you cant emote anymore and you get some "Your voice feels hoarse" flavortext when you try to

**Comment by Davidj361** [comment]
>You assume wrong
If you're gonna do this you should add the laughing sounds that are already in the game to *laugh
What do you mean? The laughing sounds are already on *laugh.

>perhaps we need a limit where if you do 3 emotes that feature sounds within a short period of time you cant emote anymore and you get some "Your voice feels hoarse" flavortext when you try to
Yeah, I think I'll add a losing voice thing, etc, but I need the maintainers to tell me how the cooldown system should be done and what types should hold these variables.

File: code/modules/mob/living/emote.dm (hunk 1 above)
Comment (Therapist1) [reply code/modules/mob/living/emote.dm]: https://www.youtube.com/watch?v=f49ELvryhao

**Comment by Iamgoofball** [comment]
why are we not just scraping the folder for file names and stitching it together

**Comment by Davidj361** [comment]
I added cooldown functionality for emotes that only have sound files involved. All of them share a cooldown of 5 seconds.
@kevinz000 @Durkel @Armhulen 

I also checked with the flu virus that you can inject yourself in virology. It only does the emote with the sound if it you're past stage 3, but yes it also forces you to do the sound emote. Viruses will now be very interesting. :smile:

File: code/modules/mob/living/carbon/emote.dm [hunk 2]
```diff
@@ -36,6 +42,28 @@
...
+/datum/emote/living/carbon/snap/run_emote(mob/living/user, params, playerInvoked = FALSE)
+	if (playerInvoked)
+		if (!isnull(cooldown_list[user.client.ckey]) && world.time < cooldown_list[user.client.ckey])
+			to_chat(user, "<span class='notice'>You cannot sound emote so soon.</span>")
+			return FALSE
+		else
+			cooldown_list[user.client.ckey] = world.time + cooldown
+	. = ..()
+	if (.)
+		if (ishuman(user))
```
Comment (optimumtact) [inline code/modules/mob/living/carbon/emote.dm]: make a subemote at the mob/living/carbon/human and check this there instead, or add a can_snap proc

File: code/modules/mob/living/emote.dm [hunk 3]
```diff
@@ -58,6 +58,28 @@
...
+
+/datum/emote/living/cough/run_emote(mob/user, params, playerInvoked = FALSE)
+	if (playerInvoked)
+		if (!isnull(cooldown_list[user.client.ckey]) && world.time < cooldown_list[user.client.ckey])
+			to_chat(user, "<span class='notice'>You cannot sound emote so soon.</span>")
+			return FALSE
+		else
+			cooldown_list[user.client.ckey] = world.time + cooldown
+	. = ..()
+	if(. && ishuman(user))
```
Comment (optimumtact) [inline code/modules/mob/living/emote.dm]: again

**Review by optimumtact — CHANGES_REQUESTED** [review]
Push the cooldown framework to the top of the emote code so it's not copied everywhere, add a generic check cooldown proc or similar

Remove all the warnings about sound not being playable, they're unnecessary and unimmersive, either the sound plays or it doesn't

push some of the human specific checks down into human typepath emotes instead.

**Review by ShizCalev — CHANGES_REQUESTED** [review]
kiss; the source sample sounds terrible. the editted version you have submitted still has a decent amount of background noise at the end of it. you'll need another file before i rubber stamp it.
mancough; the source file has a good amount of electronic distortion during the coughs.
mansneeze; way too much background noise
womancough; clip ends too abruptly, the last cough is clipping and causing crackling
womangasp; alot of background noise, honestly don't like it at all.
womanoff; tried to clean up the background noise but the clip ends too abruptly once you do
womansigh; electronic distortion which becomes even more apparent when downsampled

fixed sounds:
manoof; removed the leftover audio spike from the next part of the sample, fixed crackling
mansigh; removed background noise, trimmed silence
womansneeze; fixed clipping
womanyawn; cleaned up background noise, fixed clipping
mancough; removed background noise, trimmed silence
manyawn; removed background noise, trimmed silence
snore; just downsampled
snore2; corrected crackle at end
snapping fingers; removed the background noise in my edit
also downmixed all the fixed ones from stereo to mono

https://www.dropbox.com/sh/cqvk87lfdrgocyg/AADoBDsOMlcdGb_3xi6K0un-a?dl=0

**Comment by ShizCalev** [comment]
atomize tbh

**Comment by YakumoChen** [comment]
add comdom as a vox word please

**Comment by Davidj361** [comment]
Note to self: Make typing "lol" in chat auto-convert to *laugh.

**Comment by nicbn** [comment]
Add an evil laughter for antags when?
Also it would be nice if you added scream and burp.

**Comment by epochayur** [comment]
it's a slippery slope to *fart and *scream

**Comment by nicbn** [comment]
Yeah maybe

**Comment by ShizCalev** [comment]
stale. feel free to atomize it in another set of PRs since the vox additions are perfectly fine and would've been merged already if they were on their own.

**Pull request closed without merging**